rpc/js/src/dial.ts

```diff
@@ -209,6 +209,49 @@ interface WebRTCConnection {
 	peerConnection: RTCPeerConnection;
 }
 
+async function getOptionalWebRTCConfig(signalingAddress: string, host: string, opts?: DialOptions): Promise<WebRTCConfig> {
+    const optsCopy = { ...opts } as DialOptions;
+		const directTransport = await dialDirect(signalingAddress, optsCopy);
+
+    let pResolve: (value: WebRTCConfig) => void;
+    let pReject: (reason?: unknown) => void;
+
+    let result: WebRTCConfig | undefined;
+    let done = new Promise<WebRTCConfig>((resolve, reject) => {
+      pResolve = resolve;
+      pReject = reject;
+    });
+
+    grpc.unary(SignalingService.OptionalWebRTCConfig, {
+      request: new OptionalWebRTCConfigRequest(),
+      metadata: {
+        'rpc-host': host,
+      },
+      host: signalingAddress,
+      transport: directTransport,
+      onEnd: (resp: grpc.UnaryOutput<OptionalWebRTCConfigResponse>) => {
+        const { status, statusMessage, message } = resp;
+        if (status === grpc.Code.OK && message) {
+          result = message.getConfig();
+          if (!result) {
+            pReject("no config");
```

Comment:
I would presolve with nothing and have that be optional checked. since this is all optional, rejection feels odd (no one likes rejection!)

Reply:
sure, will just return an empty/default config in this case: https://github.com/viamrobotics/goutils/pull/157/commits/a7b5f22f45590ea2e261dd5d8bd4abedae081c41